extends Control


@onready var close_button: Button = $Button
@onready var pytanie_label: RichTextLabel = $RichTextLabel
@onready var canvas_czywieszze = %CzyWieszZeCanvas
@onready var hud = %Hud

func _ustaw_pytanie() -> void:
	var ciekawostka_text = ""

	match self.name:
		"Wiedza1":
			ciekawostka_text = "WrocÅ‚aw rozwinÄ…Å‚ siÄ™ przy ujÅ›ciu OÅ‚awy do Odry, a najstarszy OstrÃ³w Tumski zaÅ‚oÅ¼ony zostaÅ‚ na odrzaÅ„skich wyspach."
		"Wiedza2":
			ciekawostka_text = "Odra â€“ druga co do wielkoÅ›ci rzeka w Polsce bierze swÃ³j poczÄ…tek w GÃ³rach OdrzaÅ„skich w Sudetach Wschodnich na terytorium Republiki Czeskiej."
		"Wiedza3":
			ciekawostka_text = "PrzewÃ³z towarÃ³w OdrÄ… ma dÅ‚ugÄ… historiÄ™. Jeszcze w XIX wieku Å‚odzie i tratwy przepÅ‚ywaÅ‚y przez Å›rodek miasta, pokonujÄ…c liczne Å›luzy."
		"Wiedza4":
			ciekawostka_text = "OÅ‚awa wpada do Odry na PrzedmieÅ›ciu OÅ‚awskim, koÅ‚o mostu Grunwaldzkiego. Ta rzeka pÅ‚ynie takÅ¼e z wrocÅ‚awskich kranÃ³wâ€¦"
		"Wiedza5":
			ciekawostka_text = "ÅšlÄ™za do Odry wpada na styku MaÅ›lic, Pilczyc i RÄ™dzina, koÅ‚o mostu RÄ™dziÅ„skiego."
		"Wiedza6":
			ciekawostka_text = "Bystrzyca ujÅ›cie do Odry ma na Praczach OdrzaÅ„skich."
		"Wiedza7":
			ciekawostka_text = "Widawa wpada do Odry na styku RÄ™dzina i Pracz OdrzaÅ„skich."
		"Wiedza8":
			ciekawostka_text = "Nie wszystkim wiadomo, Å¼e we WrocÅ‚awiu jest w sumie nie piÄ™Ä‡, a siedem rzek. Dwie pozostaÅ‚e to Dobra i Zielona."
		"Wiedza9":
			ciekawostka_text = "Najstarszy uÅ¼ywany most we WrocÅ‚awiu to most ÅšwiÄ™tej Klary z 1799 roku, Å‚Ä…czÄ…cy WyspÄ™ MÅ‚yÅ„skÄ… i WyspÄ™ SÅ‚odowÄ…."
		"Wiedza10":
			ciekawostka_text = "Most Tumski chÄ™tnie odwiedzajÄ… szczegÃ³lnie zakochani â€“ kojarzy siÄ™ z przypinaniem kÅ‚Ã³dek i wyrzucaniem kluczyka do wody, co symbolizuje miÅ‚oÅ›Ä‡ na zawsze."
		"Wiedza11":
			ciekawostka_text = "Most RÄ™dziÅ„ski jest najdÅ‚uÅ¼szym Å¼elbetowym mostem podwieszonym w Polsce â€“ czÄ™Å›Ä‡ podwieszona ma 512 metrÃ³w dÅ‚ugoÅ›ci."
		"Wiedza12":
			ciekawostka_text = "WrocÅ‚aw to miasto mostÃ³w â€“ szacuje siÄ™, Å¼e jest ich grubo ponad 100, czyli najwiÄ™cej w Polsce."
		"Wiedza13":
			ciekawostka_text = "WrocÅ‚awski wÄ™zeÅ‚ wodny (WWW) obejmuje OdrÄ™, jej dopÅ‚ywy, kanaÅ‚y oraz urzÄ…dzenia hydrotechniczne chroniÄ…ce miasto przed powodziami."
		"Wiedza14":
			ciekawostka_text = "WWW to najwiÄ™kszy w Polsce i jeden z najwiÄ™kszych w Europie system drÃ³g wodnych i budowli hydrotechnicznych w obrÄ™bie miasta."
		"Wiedza15":
			ciekawostka_text = "Wzmianki o pierwszym porcie miejskim we WrocÅ‚awiu pochodzÄ… z 1358 roku."
		"Wiedza16":
			ciekawostka_text = "WrocÅ‚awska Gondola to statek idealny na romantyczne rejsy â€“ ma dach i zamykane boki na wypadek niepogody."
		"Wiedza17":
			ciekawostka_text = "Na poczÄ…tku XX wieku we WrocÅ‚awiu pÅ‚ywaÅ‚y gondole â€“ cumowaÅ‚y przy dzisiejszym WzgÃ³rzu Polskim, w tzw. Zatoce Gondoli."
		"Wiedza18":
			ciekawostka_text = "Hydropolis to pierwsze w Polsce centrum nauki poÅ›wiÄ™cone caÅ‚kowicie wodzie."
		"Wiedza19":
			ciekawostka_text = "SieÄ‡ wodociÄ…gowa MPWiK we WrocÅ‚awiu ma dÅ‚ugoÅ›Ä‡ aÅ¼ 2210 km i ciÄ…gle siÄ™ rozrasta."
		"Wiedza20":
			ciekawostka_text = "Woda dostarczana mieszkaÅ„com WrocÅ‚awia pochodzi z rzeki OÅ‚awy i Nysy KÅ‚odzkiej â€“ trafia do ZUW â€žMokry DwÃ³râ€ i â€žNa Grobliâ€."
		"Wiedza21":
			ciekawostka_text = "Czy wiesz, Å¼e Odra jest drugÄ… co do dÅ‚ugoÅ›ci rzekÄ… w Polsce, a jej dolnoÅ›lÄ…ski odcinek byÅ‚ wielokrotnie regulowany, by zapobiegaÄ‡ powodziom?"
		"Wiedza22":
			ciekawostka_text = "Czy wiesz, Å¼e WrocÅ‚aw byÅ‚ wielokrotnie nawiedzany przez powodzie, a najwiÄ™ksza z nich w 1997 roku nazwana zostaÅ‚a â€žpowodziÄ… tysiÄ…cleciaâ€?"
		"Wiedza23":
			ciekawostka_text = "Czy wiesz, Å¼e aÅ¼ 75% sÅ‚odkiej wody jest uwiÄ™zione w lodowcach i pokrywach lodowych?"
		"Wiedza24":
			ciekawostka_text = "Czy wiesz, Å¼e wiÄ™kszoÅ›Ä‡ polskich rzek naleÅ¼y do zlewiska BaÅ‚tyku, ale sÄ… teÅ¼ takie, ktÃ³re wpadajÄ… do Morza Czarnego i PÃ³Å‚nocnego."
		"Wiedza25":
			ciekawostka_text = "Czy wiesz, Å¼e najgÅ‚Ä™bsze jezioro Å›wiata to BajkaÅ‚, ktÃ³rego gÅ‚Ä™bokoÅ›Ä‡ wynosi 1642 m?"
		"Wiedza26":
			ciekawostka_text = "Czy wiesz, Å¼e w Dolinie KÅ‚odzkiej znajdujÄ… siÄ™ jedne z najbardziej tajemniczych podziemnych rzek krasowych w Polsce?"
		"Wiedza27":
			ciekawostka_text = "Czy wiesz, Å¼e wody mineralne z Polanicy-Zdroju, Kudowy-Zdroju i LÄ…dka-Zdroju sÄ… znane w caÅ‚ej Polsce i majÄ… lecznicze wÅ‚aÅ›ciwoÅ›ci?"
		"Wiedza28":
			ciekawostka_text = "Czy wiesz, Å¼e w GÃ³rach Sowich moÅ¼na znaleÅºÄ‡ Å›lady dawnych jezior polodowcowych, ktÃ³re zanikÅ‚y na skutek erozji?"
		"Wiedza29":
			ciekawostka_text = "Czy wiesz, Å¼e pod WrocÅ‚awiem istniejÄ… podziemne korytarze wodne, ktÃ³re byÅ‚y wykorzystywane do obrony miasta w dawnych czasach?"
		"Wiedza30":
			ciekawostka_text = "Czy wiesz, Å¼e na Dolnym ÅšlÄ…sku znajdujÄ… siÄ™ unikalne formy moren czoÅ‚owych, ktÃ³re sÄ… pozostaÅ‚oÅ›ciÄ… po lodowcu?"

	var ciekawostka_template = """
[center][font_size=30][b][color=#FFD700]ðŸŒŸ Czy wiesz, Å¼e... ðŸŒŸ[/color][/b][/font_size]

[font_size=25][color=#FFFFFF]""" + ciekawostka_text + """[/color][/font_size][/center]
"""
	pytanie_label.text = ciekawostka_template
			
func _ready() -> void:
	pytanie_label.bbcode_enabled = true
	_ustaw_pytanie()
	canvas_czywieszze.visible = false
	visible = false
	close_button.pressed.connect(_on_close_pressed)

# Function to show the entire popup window
func show_popup() -> void:
	canvas_czywieszze.visible = true
	visible = true

# Function to hide the entire popup window
func _on_close_pressed() -> void:
	canvas_czywieszze.visible = false
	visible = false
	hud.increase_score(1)
	

func toggle_visible() -> void:
	canvas_czywieszze.visible = !canvas_czywieszze.visible
	visible = !visible
